### PR TITLE
Microsoft Search: KQL query text Extension

### PR DESCRIPTION
**Pull request by edarroudi**
Added a property 'Query modifier in Microsoft Search DataSource 
image

Added labels for the different languages through google translate.

It just attaches whatever is defined in the property to the queryinput text.
e.g. if template is set to 'AND IsDocument:true',  a search with the input string 'security' will be transformed to 'security AND IsDocument:true'

The template accepts tokens as done in SharePointSearch:QuerySearchTemplate.

Was discussed in issue : https://github.com/microsoft-search/pnp-modern-search/issues/1467

### CODE REVIEW

**Comment by wobba** [comment]
I think you should omit `AND` from the sample, as AND is implicit anyways in KQL.

Curious if we should add `{searchTerms}` as a concept and then replace it with the user query if present - to make it consistent from a web part point of view.

Any comments @FranckyC?

**Comment by edarroudi** [comment]
> I think you should omit `AND` from the sample, as AND is implicit anyways in KQL.
> 
> Curious if we should add `{searchTerms}` as a concept and then replace it with the user query if present - to make it consistent from a web part point of view.
> 
> Any comments @FranckyC?

I did have it like this in my earlier version, but as {searchTerms} isn't part of the tokenService and i had just a regex replace for it in code it looked out of place. 

I'll remove the ADD part, forgot the implicity, thanks.

**Comment by edarroudi** [comment]
found a setting error then in context if we go the way right now without {searchTerms}. allowEmptyValue has to be true.
I will fix this and the rest when you guys decicde what your want :)

**Comment by edarroudi** [comment]
Hey @wobba, 
did you decide what you want: either including {searchTerms} in the modifier template or as it is right now, concat it to the inputModifier. Please tell me if i can do something else to get this approved ;-)

**Comment by pettersh** [comment]
In my opinion I think the property label should be **Query template** instead of **Query modifier** for consistency. 

I understand that this feature is a query modification, but that's what SharePoint query templates does under the hood as well; modifies the user's query before it's sent to the search index.

**Comment by edarroudi** [comment]
@pettersh If i rename it to Query template then i have to go the way @wobba described, include {searchTerms} in the Query template - for consitency reasons :) I think that would be better anyways for overall consistency.

**Comment by wobba** [comment]
@edarroudi did you implement my ask? I think end-user familiarity is more important than the actual technical implementation. It also allows easier migration from SP API to Graph API.

**Comment by edarroudi** [comment]
@wobba  sorry - didn'T see your thumbs up. I'll do it till monday.

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 1]
```diff
@@ -468,12 +502,19 @@ export class MicrosoftSearchDataSource extends BaseDataSource<IMicrosoftSearchDa
...
         // Query text
         if (dataContext.inputQueryText) {
             queryText = await this._tokenService.resolveTokens(dataContext.inputQueryText);
         }
 
+        // Query modification
+        const queryModifierTemplate = await this._tokenService.resolveTokens(this.properties.queryModifierTemplate);
+        if(queryModifierTemplate)
+        {
+            queryText = `${queryText} ${queryModifierTemplate.trimLeft()}`;
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:515]: Wrong statement, you should have the same behavior as the SharePoint search. The query text should be included on demand using either `{searchTerms}` and `{inputQueryText}` tokens. It means you should add `{searchTerms}` as a builtin token and resolve it the same way as `{inputQueryText}`.

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 2]
```diff
@@ -59,7 +60,14 @@ export interface IMicrosoftSearchDataSourceProperties {
     contentSourceConnectionIds: string[];
 
     /**
-     * Flag indicating if the Microsoft Search beta endpoint should be used
+     * The search modfier template. Extends the query with additional KQL
+     * e.g. if tempalte is set to 'IsDocument:true' a search with the input string 'security'
+     * will be transformed to 'security IsDocument:true'
+     */
+     
+    queryModifierTemplate: string;
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:68]: Rename to `queryTemplate` to align with SharePoint Search

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 3]
```diff
@@ -451,6 +483,8 @@ export class MicrosoftSearchDataSource extends BaseDataSource<IMicrosoftSearchDa
         this.properties.fields = this.properties.fields !== undefined ? this.properties.fields : CommonFields;
         this.properties.sortProperties = this.properties.sortProperties !== undefined ? this.properties.sortProperties : [];
         this.properties.contentSourceConnectionIds = this.properties.contentSourceConnectionIds !== undefined ? this.properties.contentSourceConnectionIds : [];
+        
+        this.properties.queryModifierTemplate = this.properties.queryModifierTemplate ?? "";
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:487]: Use the same initialization as SharePoint Search `this.properties.queryTemplate = this.properties.queryTemplate ? this.properties.queryTemplate : "{searchTerms}";`

**Review by FranckyC — CHANGES_REQUESTED** [review]
Please update according to comments before merging. Also check with the current `develop` branch and add missing translations if any. Thanks!

**Comment by FranckyC** [comment]
Hi @edarroudi, I've made the changes in the PR #1621 because I needed them before the December release. I'll close this one.

**Pull request closed without merging**